src/parser/helper/TreeParser.ts

```diff
@@ -43,10 +23,27 @@ export class TreeParser {
             return cachedItem;
         }
 
+        const sourceCode = await fs.promises.readFile(parseFile.filePath, { encoding: "utf8" });
+
+        return TreeParser.parse(sourceCode, parseFile);
+    }
+
+    private static parse(sourceCode: string, parseFile: ParseFile) {
+        let grammarLanguage = parseFile.language;
+
+        if (grammarLanguage === Languages.JavaScript) {
+            // Check if this is actually flow-annotated code instead of plain JavaScript. Use the TSX-grammar then.
+            // See https://flow.org/en/docs/usage/#toc-prepare-your-code-for-flow on how to identify them.
+            // See https://github.com/tree-sitter/tree-sitter-typescript/tree/v0.20.5 on using the TSX-grammar
+            // for flow-annotated files.
+            if (sourceCode.match(/^.*@flow/) !== null) {
```

Comment:
I am surprised. Can there be files without .tsx extension that are written in tsx code?

Reply:
"For Javascript files with [flow](https://flow.org/en/) type annotations you can use the the tsx parser."
Source: https://github.com/tree-sitter/tree-sitter-typescript

Reply:
According to the flow documentation [linked here](https://flow.org/en/docs/usage/#toc-prepare-your-code-for-flow), flow-files can be differentiated from plain JavaScript-files via an annotiation in a file header comment.

Reply:
As far as I know, tsx code is something different again ([see here](https://www.typescriptlang.org/docs/handbook/jsx.html)). But the tsx-grammar of tree-sitter-typescript can be also used to parse flow-annotated JavaScript.